planner/optimize.go

```diff
@@ -88,10 +96,13 @@ func Optimize(ctx context.Context, sctx sessionctx.Context, node ast.Node, is in
 			bestPlanAmongHints = plan
 		}
 	}
+	// If there is already a evolution task, we do not need to handle it again.
+	if sctx.GetSessionVars().EvolvePlanBaselines && binding == nil {
```

Comment:
If don't want to evolve baseline or there's existing binding, can we just break and return here?

Reply:
No, it needs to handle the case of `UsePlanBaselines`.